app/frontend/wallet/shelley/shelley-transaction-planner.ts

```diff
@@ -149,24 +150,45 @@ function computeRequiredDeposit(certificates: Array<TxCertificate>): Lovelace {
   return certificates.reduce((acc, {type}) => acc + CertificateDeposit[type], 0) as Lovelace
 }
 
-export const computeMinUTxOLovelaceAmount = (
-  address: Address,
-  coins: Lovelace,
-  tokens: Token[]
-): Lovelace => {
+export const computeMinUTxOLovelaceAmount = (tokenBundle: TokenBundle): Lovelace => {
+  // based on https://github.com/input-output-hk/cardano-ledger-specs/blob/master/doc/explanations/min-utxo.rst
+  const quot = (x: number, y: number) => Math.floor(x / y)
+  const roundupBytesToWords = (x: number) => quot(x + 7, 8)
   // TODO: this to network config or constants
-  const adaOnlyUTxOSize = 74 // TODO: this is too much
   const minUTxOValue = 1000000
-  if (tokens.length === 0) return minUTxOValue as Lovelace
-  const output: TxOutput = {
-    isChange: false,
-    address,
-    coins,
-    tokens,
+  // NOTE: should be 2, but a bug in Haskell set this to 0
+  const coinSize = 0
+  const txOutLenNoVal = 14
+  const txInLen = 7
+  const utxoEntrySizeWithoutVal = 6 + txOutLenNoVal + txInLen // 27
+
+  // NOTE: should be 29 but a bug in Haskell set this to 27
```

Comment:
would be nice to reference the code/issue as this seems pretty obscure and hard to track later on

Reply:
It would be nice to reference the issue but I think its sufficient enough for now. Who knows if there even is an issue for this, I think the time spent looking for the cause in cardano node respo or looking through all the issues in that repository is not worth it imo.